lib/workers/repository/reconfigure/index.ts

```diff
@@ -17,15 +33,114 @@ export async function checkReconfigureBranch(
     return;
   }
 
+  const context = config.statusCheckNames?.configValidation;
   const reconfigureBranch = getReconfigureBranchName(config.branchPrefix!);
   const branchExists = await scm.branchExists(reconfigureBranch);
 
-  // this is something the user initiates, so skip if no branch exists
+  // this is something the user initiates, skip if no branch exists
   if (!branchExists) {
     logger.debug('No reconfigure branch found');
     deleteReconfigureBranchCache(); // in order to remove cache when the branch has been deleted
     return;
   }
 
-  await validateReconfigureBranch(config);
+  const existingPr = await platform.findPr({
+    branchName: reconfigureBranch,
+    state: 'open',
+    includeOtherAuthors: true,
+    targetBranch: config.defaultBranch,
+  });
+  const branchSha = getBranchCommit(reconfigureBranch)!;
+  const cache = getCache();
+  const reconfigureCache = cache.reconfigureBranchCache;
+
+  // only use valid cached information
+  if (reconfigureCache?.reconfigureBranchSha === branchSha) {
+    if (!existingPr || reconfigureCache.extractResult) {
+      logger.debug('Skipping validation check as branch sha is unchanged');
+      return;
+    }
+  }
+
+  const result = await getReconfigureConfig(reconfigureBranch);
+
+  if (!result.ok) {
+    await setBranchStatus(reconfigureBranch, result.errMessage, 'red', context);
+    setReconfigureBranchCache(branchSha, false);
+    await scm.checkoutBranch(config.defaultBranch!);
+    return;
+  }
+
+  const isValidConfig = await validateReconfigureBranch(
+    config,
+    result.config,
+    result.configFileName,
+    existingPr,
+  );
+
+  if (!isValidConfig) {
+    logger.debug(
```

Comment:
set branch status here? or is it done in `validateReconfigureBranch` ?

Reply:
done in `validateReconfigureBranch`